subsys/net/ip/net_if.c

```diff
@@ -518,11 +523,12 @@ enum net_verdict net_if_send_data(struct net_if *iface, struct net_pkt *pkt)
 		}
 	} else if (verdict == NET_OK) {
 		/* Packet is ready to be sent by L2, let's queue */
+		net_if_unlock(iface);
```

Comment:
Is this really needed? Mutex lock shouldn't block if called from the same thread.

Reply:
Good point, Zephyr mutexes are re-entrant, will remove.